File: ScalaExperimente/src/adressverwaltung/Address.scala
package adressverwaltung

class Address(val street: String, val city: String) {
	
  def contains(part: String) = street.contains(part) || city.contains(part)

  override def equals(other: Any) = other match {
    case that: Address => this.street == that.street && this.city == that.city
    case _ => false
  }

  override def toString = street + ", " + city

  override def hashCode = street.hashCode + city.hashCode
}

File: JavaExperimente/src/adressverwaltung/Address.java
package adressverwaltung;

public class Address {

	private final String street;
	private final String city;

	public Address(String street, String city) {
		this.street = street;
		this.city = city;
	}

	@Override
	public boolean equals(Object that) {
		return that instanceof Address && equals((Address) that);
	}

	private boolean equals(Address that) {
		return this.street.equals(that.street) & this.city.equals(that.city);
	}

	@Override
	public int hashCode() {
		return street.hashCode();
	}

	@Override
	public String toString() {
		return street + ", " + city;
	}
}
